### PR TITLE
improve compatibility with eslint@9 (v2)

### LINKED ISSUE #30: ESLint v9 Support
ESLint has released v9: https://eslint.org/blog/2024/04/eslint-v9.0.0-released/

Could you please add compatibility with version 9?

As far as I can see, there are quite a lot of breaking changes there. And flat format is now enabled by default. I'm not sure, but this means that you either need to migrate completely to a flat config, or maintain two versions of the config.

### PR DESCRIPTION
**Pull request by dartess**
Closed https://github.com/remithomas/eslint-plugin-ban/issues/30

So, now we are aiming for the major.

The lint of the project itself has also been converted to flat config.

Additionally, we can limit the version of `eslint` in `peerDeps` and the version of `node` in `engines`, however, I don't see the point in this now. If someday in the future there are _really_ major changes that break backwards compatibility, it will be easier to release a new major. For now, I suggest leaving a simple warning in README. But that's up for debate.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,9 +17,8 @@
   "author": "Rémi THOMAS",
   "main": "lib/index.js",
   "scripts": {
-    "lint": "eslint .",
-    "lint:fix": "eslint . --fix",
-    "format": "prettier --write 'lib/*.js' 'tests/**/*.js'",
```
Comment (dartess) [inline package.json:22]: duplicates the work of lint:fix

File: package.json [hunk 2]
```diff
@@ -28,14 +27,14 @@
     "requireindex": "~1.2.0"
   },
   "devDependencies": {
-    "eslint": "^8.2.0",
-    "eslint-config-prettier": "^8.3.0",
-    "eslint-plugin-prettier": "^4.0.0",
-    "growl": "^1.10.5",
```
Comment (dartess) [inline package.json:34]: not used

File: package.json [hunk 3]
```diff
@@ -28,14 +27,14 @@
...
+    "@eslint/js": "^9.12.0",
+    "eslint": "^9.12.0",
+    "eslint-config-prettier": "^9.1.0",
+    "eslint-plugin-prettier": "^5.2.1",
+    "globals": "^15.11.0",
     "husky": "^7.0.4",
-    "jest": "^27.3.1",
```
Comment (dartess) [inline package.json:36]: not used

File: tests/lib/rules/ban.js [hunk 4]
```diff
@@ -196,19 +196,6 @@ ruleTester.run('ban with multiple rules', rule, {
       ],
       options: multipleRuleOptions,
     },
-    {
-      code: 'functionName("dogs");',
```
Comment (dartess) [inline tests/lib/rules/ban.js:200]: duplicated, which is prohibited by eslint@9

**Review by remithomas — APPROVED** [review]

**Pull request merged**